### PR TITLE
planner: update the variable name related to Instance Plan Cache

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #54057

Problem Summary: planner: update the variable name related to Instance Plan Cache

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -1348,19 +1348,13 @@ var defaultSysVars = []*SysVar{
 			EnableInstancePlanCache.Store(TiDBOptOn(val))
 			return nil
 		}},
-	{Scope: ScopeGlobal, Name: TiDBInstancePlanCacheTargetMemSize, Value: strconv.Itoa(int(DefTiDBInstancePlanCacheTargetMemSize)), Type: TypeStr,
+	{Scope: ScopeGlobal, Name: TiDBInstancePlanCacheReservedPercentage, Value: "0.1", Type: TypeFloat, MinValue: 0, MaxValue: 1,
```
Comment (AilinKid) [inline pkg/sessionctx/variable/sysvar.go:1351]: why not DefInstancePlanCacheReservedPercentage = “0.1”

**Review by AilinKid — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

**Review by yudongusa — APPROVED** [review]
LGTM for the changes.

**Pull request merged**